### ISSUE TITLE
Problem with character encoding for "≥"

### ISSUE THREAD

**Issue by rbroth** [sqlfluff/sqlfluff#1130]
I get an error when running sqlfluff against a file that has greater-than-or-equal-to symbol in one of the column comments. It seems to be some sort of encoding problem. Could also be related to me being on windows.

## Expected Behaviour

`sqlfluff  fix  'D:\fluff.sql'` should fix the file

## Observed Behaviour

Running sqlfluff fix on a file containing a `≥` character deletes the contents of the file.

## Steps to Reproduce
Create file fluff.sql. (Using VSCode on windows. Encoding is set to UTF-8)
```sql
CREATE TABLE table (
    , col                     numeric(3,2)
);
COMMENT ON COLUMN table.col                 IS ' ≥1 ';
```

### sqlfluff fix
```
$ sqlfluff fix 'D:\fluff.sql'
==== finding fixable violations ====
== [D:\fluff.sql] FAIL
L:   2 | P:   5 | L019 | Found leading comma. Expected only trailing.
L:   2 | P:  10 | L039 | Unnecessary whitespace found.
L:   2 | P:  41 | L008 | Commas should be followed by a single whitespace unless
                       | followed by a comment.
L:   4 | P:  28 | L039 | Unnecessary whitespace found.
==== fixing violations ====
4 fixable linting violations found
Are you sure you wish to attempt to fix these? [Y/n] ...
Attempting fixes...
Persisting Changes...
Traceback (most recent call last):
  File "c:\program files\python39\lib\runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "c:\program files\python39\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Users\redacted\AppData\Roaming\Python\Python39\Scripts\sqlfluff.exe\__main__.py", line 7, in <module>
  File "C:\Users\redacted\AppData\Roaming\Python\Python39\site-packages\click\core.py", line 1137, in __call__
    return self.main(*args, **kwargs)
  File "C:\Users\redacted\AppData\Roaming\Python\Python39\site-packages\click\core.py", line 1062, in main
    rv = self.invoke(ctx)
  File "C:\Users\redacted\AppData\Roaming\Python\Python39\site-packages\click\core.py", line 1668, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "C:\Users\redacted\AppData\Roaming\Python\Python39\site-packages\click\core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "C:\Users\redacted\AppData\Roaming\Python\Python39\site-packages\click\core.py", line 763, in invoke
    return __callback(*args, **kwargs)
  File "C:\Users\redacted\AppData\Roaming\Python\Python39\site-packages\sqlfluff\cli\commands.py", line 462, in fix
    success = do_fixes(
  File "C:\Users\redacted\AppData\Roaming\Python\Python39\site-packages\sqlfluff\cli\commands.py", line 360, in do_fixes
    res = result.persist_changes(formatter=formatter, **kwargs)
  File "C:\Users\redacted\AppData\Roaming\Python\Python39\site-packages\sqlfluff\core\linter.py", line 776, in persist_changes
    *[
  File "C:\Users\redacted\AppData\Roaming\Python\Python39\site-packages\sqlfluff\core\linter.py", line 777, in <listcomp>
    path.persist_changes(formatter=formatter, **kwargs)
  File "C:\Users\redacted\AppData\Roaming\Python\Python39\site-packages\sqlfluff\core\linter.py", line 632, in persist_changes
    buffer[file.path] = file.persist_tree(suffix=fixed_file_suffix)
  File "C:\Users\redacted\AppData\Roaming\Python\Python39\site-packages\sqlfluff\core\linter.py", line 552, in persist_tree
    f.write(write_buff)
  File "c:\program files\python39\lib\encodings\cp1252.py", line 19, in encode
    return codecs.charmap_encode(input,self.errors,encoding_table)[0]
UnicodeEncodeError: 'charmap' codec can't encode character '\u2265' in position 83: character maps to <undefined>
```

### sqlfluff parse
```
$ sqlfluff parse 'D:\fluff.sql'
...Failed to Parse...
```

### sqlfluff lint
```
$ sqlfluff lint 'D:\fluff.sql'

```

## Version

sqlfluff: 0.5.6
python: 3.9.5
Using VSCode integrated bash terminal on Windows 10 Enterprise
Code is for postgres 12

## Configuration
No custom config, using it out-of-the box after first install

**Comment by barrywhart** [comment]
Is this operator supported in Postgres itself? Is this documented somewhere?

**Comment by tunetheweb** [comment]
I presume since it's a comment anything should be allowed? Is that the request here @rbroth ?

**Comment by rbroth** [comment]
> I presume since it's a comment anything should be allowed? Is that the request here @rbroth ?

Correct, my expectation would be that anything that can be put into a PostgreSQL string (including wierd unicode characters) shouldn't break sqlfluff.

**Comment by rbroth** [comment]
> Is this operator supported in Postgres itself? Is this documented somewhere?

Sorry for not seeing this before. In this case `≥` is not being used as an operator. It's just the contents of a column comment, which is a string literal. Postgres supports a cariaet

From what I can tell from the error message, I suspect that there is some problem with character encoding. My code file is in UTF-8, but sqlfluff expects all input to be in in windows cp1252.

**Comment by tunetheweb** [comment]
Yeah I don't get an issue with it on a Mac. It does print incorrectly on `sqlfluff parse` but no error. And `sqlfluff fix` does fix it fine.

So definitely some Windows specific thing.

Can you try with the latest version of sqlfluff (0.6.6)? As someone else had a similar issue that was resolved since they raised it.

**Comment by rbroth** [comment]
> Yeah I don't get an issue with it on a Mac. It does print incorrectly on `sqlfluff parse` but no error. And `sqlfluff fix` does fix it fine.
> 
> So definitely some Windows specific thing.
> 
> Can you try with the latest version of sqlfluff (0.6.6)? As someone else had a similar issue that was resolved since they raised it.

Yup, sqlfluff installed via pip works fine. The vs-code version is still on v0.0.5, so I assume that whatever was updated in the meantime must have fixed it. 

Thank you very much for your help. 

Is there a timeframe for when to expect the vscode extension to be updated?

**Comment by tunetheweb** [comment]
I thought the VS Code version just used whatever you’d installed locally?

**Comment by tunetheweb** [comment]
From here: https://marketplace.visualstudio.com/items?itemName=dorzey.vscode-sqlfluff

> Configuration
> 
> The extension expects sqlfluff to be installed and already added to the path. If it is installed but cannot be found, add the path to your preferences as seen below. Find the path by typing which sqlfluff into your terminal.

**Comment by rbroth** [comment]
oof, I honestly have no idea. I might have installed it and forgotten about it. The python environment on my machine is... a bit of a mess. I'll have a look at some point. In the meantime, feel free to close this issue, and thanks again for your help.

**Comment by rbroth** [comment]
Oh, or I mistook the version of the extension wrapper for the sqlfluff version. Derp. Time to go to bed
